lambdas/es/indexer/index.py

```diff
@@ -197,6 +197,9 @@ def infer_extensions(key, ext):
             or ext == ".pq"
     ):
         return ".parquet"
+    # if there are multiple dots take only the last suffix
+    elif ext.count(".") > 1:
+        return pathlib.PurePosixPath(ext).suffix
```

Comment:
I think this is the wrong place to handle it just cause we already have code that is supposed to handle multiple extensions: https://github.com/quiltdata/quilt/pull/3446/files#diff-75e3588577c6f019a82536d2c88e59ca3d77cb035b1d933e8f3d92fda26b8c67R705-R708

I'm not sure what what the best solution is, but we could:
- pass `ext1` and `ext2` instead of `ext`, OR
- move the `.gz` handling up the stack, and guarantee that `ext` is only a single extension